lms/djangoapps/course_api/blocks/api.py

```diff
@@ -61,6 +64,16 @@ def get_blocks(
     # transform
     blocks = get_course_blocks(user, usage_key, transformers)
 
+    # filter blocks by types
+    if block_types:
+        block_keys_to_remove = []
+        for block_key in blocks:
+            block_type = blocks.get_xblock_field(block_key, 'category')
+            if block_type not in block_types:
+                block_keys_to_remove.append(block_key)
+        for block_key in block_keys_to_remove:
+            blocks.remove_block(block_key, True)
```

Comment:
For code readability, please add the keyword name to the boolean parameter: `keep_descendants=True`.